zendesk_sla_metrics/sla_response_time_targets

```diff
@@ -1,103 +1,215 @@
+  join pause_times on start_time_in_minutes.ticket_id = pause_times.ticket_id
+    and start_time_in_minutes.metric = pause_times.metric 
+    and (
+      (timestamp_add(start_of_week,interval start_time_utc minute) <= pause_time  and timestamp_add(start_of_week,interval end_time_utc minute) >= resume_time )
+      or 
+      (timestamp_add(start_of_week,interval start_time_utc minute) between pause_time  and resume_time )
+      or 
+      (timestamp_add(start_of_week,interval end_time_utc minute) between pause_time  and resume_time )
+    ) 
 ), time_left as (
 -- track time worked against the target to calculate breach time
   select 
-   *,
-   target - sum(end_time_utc-if(end_time_utc +(week_number*(7*24*60)) > start_time, greatest(start_time_utc,start_time),end_time_utc)) over running_total as time_left
-  from start_time_in_minutes
+    *,
+    timestamp_add(start_of_week,interval start_time_utc minute) start_actual, timestamp_add(start_of_week,interval end_time_utc minute) end_actual,
+    target - sum((end_time_utc -if(end_time_utc +(week_number*(7*24*60)) > start_time, greatest(start_time_utc,start_time),end_time_utc) )*mult) over running_total as time_left
+  from adding_pause_times
   window running_total as (partition by ticket_id, metric, instance_id order by week_number,start_Time_utc rows between unbounded preceding and current row)
 ), breach_time as (
 -- find breach by looking at min negative time_left value, and adding the time back to the schedule end time
   select
-   ticket_id,
-   metric,
-   instance_id,
-   sla_policy_Id,
-   schedule_id,
-   min(timestamp_add(start_of_week,interval end_time_utc+time_left.time_left minute)) as breach_time
+    ticket_id,
+    metric,
+    instance_id,
+    sla_policy_Id,
+    schedule_id,
+    min(timestamp_add(start_of_week,interval end_time_utc+time_left.time_left minute)) as breach_time
   from time_left
-  where time_left.time_left<0
+  where time_left.time_left<=0
   group by 1,2,3,4,5
 )
 
 select 
- breach_time.ticket_id,
- breach_time.metric,
- breach_time.instance_id,
- breach_time.sla_policy_id,
- breach_time.schedule_id,
- sla_metric_by_ticket.sla_applied_at sla_start_time,
- if(business_hours,breach_time.breach_time,timestamp_add(sla_applied_at, interval target minute)) as breach_time,
- sla_metric_by_ticket.target,
- if(ticket_first_responded_at.responded_at is null, true, false) as is_active
+  breach_time.ticket_id,
+  breach_time.metric,
+  breach_time.instance_id,
+  breach_time.sla_policy_id,
+  breach_time.schedule_id,
+  sla_metric_by_ticket.sla_applied_at sla_start_time,
+  if(business_hours,breach_time.breach_time,timestamp_add(sla_applied_at, interval target minute)) as breach_time,
+  sla_metric_by_ticket.target,
+  case 
+    when breach_time.metric = 'first_reply_time' and ticket_first_responded_at.responded_at is null then true
+    when breach_time.metric in ('agent_work_time', 'requester_wait_time') and open_tickets.id is not null then true
```

Comment:
not sure how we want to address next reply time here. We would need to calculate all the responses on a ticket and compare the instances to the responses